### ISSUE TITLE
problem with csync_vio_stat for long Korean names

### ISSUE THREAD

**Issue by moscicki** [owncloud/client#6638]
Seen with client 2.3.3 on MacOSX. Reported by a user but I cannot reproduce it.

Some directories are not synchronized, as shown in this screenshot:

![image](https://user-images.githubusercontent.com/621965/42377341-4796ff20-8123-11e8-9512-f65c533e7081.png)

Inspecting the client log I see several csync_vio_stat failure with EFAULT (14) and ENOENT(2):

```
10-29 18:25:57:071 0x7f95a270b5d8 _csync_detect_update: file: KNU/연구/2017년도 상반기 EU 인력교류 프로그램(MSCA ITN) 참여 희망기관 수요조사/[붙임1] 2017년도 상반기 MSCA ITN 인력교류 프로그램 참여 희망기관 수요조사.hwp, instruction: INSTRUCTION_NONE <<=
10-29 18:25:57:071 0x7f95a270b5d8 csync_vio_stat: Local stat failed, errno 14
--
10-29 18:25:57:072 0x7f95a270b5d8 csync_ftw:  <= Closing walk for /Users/csm/Cernbox/KNU/연구/2017년도 상반기 EU 인력교류 프로그램(MSCA ITN) 참여 희망기관 수요조사 with read_from_db 0
10-29 18:25:57:072 0x7f95a270b5d8 csync_vio_stat: Local stat failed, errno 2
10-29 18:25:57:072 0x7f95a270b5d8 csync_vio_stat: Local stat failed, errno 2
--
10-29 18:25:57:072 0x7f95a270b5d8 _csync_detect_update: KNU/연구/참고/2016년도「일반 연구교류지원사업」신규 공고 (자주하는 질문 첨부)/.DS_Store excluded  (2)
10-29 18:25:57:072 0x7f95a270b5d8 csync_vio_stat: Local stat failed, errno 14
```

Are there known issues with long unicode characters?

Here is a zipped archive which contains a hierarchy of such filenames and directories for testing:

[temp2.zip](https://github.com/owncloud/client/files/2170380/temp2.zip)

**Comment by ogoffart** [comment]
There were unicode problem before on macosx. Most have been fixed in 2.4.  (And maybe some in 2.5)
Please try again with 2.4.1,  I think this particular issue was fixed.

**Comment by ogoffart** [comment]
Related:  #5661

**Comment by guruz** [comment]
> Please try again with 2.4.1

Or even 2.4.2 rc1
https://central.owncloud.org/t/desktop-sync-client-2-4-2-rc1-released-release-candidate/14408

**Comment by guruz** [comment]
@moscicki 2.4.2 is out :)

**Comment by ogoffart** [comment]
Assumed to be fixed in 2.4.x
